### DISCUSSION TITLE
Torch reports CUDA not available

### DISCUSSION THREAD

**Discussion by LineKernel** [bmaltais/kohya_ss#911] [Q&A] [answered]
hi 
At startup i get the "Torch reports CUDA not available" warning , i m running on a 1080ti .Right now i successfully config Kohya to train lora , but it s extremely slow .Will having CUDA working make training faster ? (i was thinking if it s not using CUDA it's going though the CPU , am i right ?)

i went and downloaded cuda_12.1.1_531.14_windows.exe , i installed most of the things but didnt install the display drivers . 
do i need to install the display drivers too ?

if not, how do i refresh Torch ? cause the warning is still here 

another question , is Torch dependent on the Python install and reside in the python folder and is common for everything , or per project , like , is there a torch in my a1111 and another torch in my kohya folder ?

thanks

**Comment by LineKernel** [comment]
i did the update from yesterday :
nVidia toolkit detected
Torch 2.0.1+cpu
WARNING  Torch reports CUDA not available
Verifying requirements
INFO     headless: False

**Reply by LineKernel** [reply]
now i get 
AttributeError: module 'tensorflow' has no attribute 'io'
when i try to train

**Comment by LineKernel** [comment]
ok , i m trying from scratch with CUDA 11.8

**Reply by LineKernel** [reply]
pip install torch==2.0.1+cu118 torchvision==0.15.1+cu118 --extra-index-url https://download.pytorch.org/whl/cu118

now i have to update xformers to version 0.0.20

**Comment by LineKernel** [comment, marked as answer]
ok , with the right cuda (11.8), and the right xformers(0.0.20) , clean install , the good accelerate config , it starts with the GPU this time

**Reply by LineKernel** [reply]
ok with torch 2.0.1  i should NOT use bf16 everything else works

**Comment by LearnTheNext** [comment]
Really appreciate you posting the resolution including pip code. I had a dependency conflict when using the listed versions, so I had to update the torchversion to work with torch 2.0.1.

pip install torch==2.0.1+cu118 torchvision==0.15.2+cu118 --extra-index-url https://download.pytorch.org/whl/cu118

**Comment by DNG-hub** [comment]
This approach did not work for me.  However, as part of the error message I received instructed "The conflict is caused by:
    The user requested torch==2.0.1+cu118
    torchvision 0.15.1+cu118 depends on torch==2.0.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict"         I found the answer on GitHub "Version solving fails with PyTorch +cu118 #7876"        the proper code that worked for me was was linked in the article https://pytorch.org/get-started/locally/         the pip command is  "pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118"

**Comment by DNG-hub** [comment]
even though the the above process allowed everything to load and the the Browser interface launched, I still received this error:
21:20:25-110077 INFO     nVidia toolkit detected
21:20:27-013894 INFO     Torch 2.1.0+cpu.
21:20:27-013894 WARNING  Torch reports CUDA not available
21:20:27-014879 INFO     Verifying modules installation status from requirements_windows_torch2.txt...
21:20:27-015888 INFO     Verifying modules installation status from requirements.txt...
21:20:27-018458 WARNING  Package wrong version: huggingface-hub 0.17.3 required 0.15.1
21:20:27-019458 INFO     Installing package: huggingface-hub==0.15.1
21:20:29-743887 INFO     headless: False
21:20:29-745885 INFO     Load CSS...
Running on local URL:  http://127.0.0.1:7860/

I went through many possible solutions, like configuring CUDA_HOME and installing the transformers to the correct version,I still got the same error.  It now works by finally doing this:
pip uninstall torch
pip install torch==2.1.0+cu118 -f https://download.pytorch.org/whl/torch_stable.html

I now have my gpu
16:27:04-727945 INFO     nVidia toolkit detected
16:27:05-621951 INFO     Torch 2.1.0+cu118
16:27:05-633427 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
16:27:05-635344 INFO     Torch detected GPU: NVIDIA GeForce RTX 4090 VRAM 24564 Arch (8, 9) Cores 128
16:27:05-636364 INFO     Verifying modules installation status from requirements_windows_torch2.txt...
16:27:05-637356 INFO     Verifying modules installation status from requirements.txt...
16:27:07-932207 INFO     headless: False
16:27:07-934214 INFO     Load CSS...
Running on local URL:  http://127.0.0.1:7860

Hope this helps others
